addons/mail/static/src/discuss/gif_picker/common/gif_picker.js

```diff
@@ -175,15 +175,22 @@ export class GifPicker extends Component {
             if (this.next) {
                 params.position = this.next;
             }
-            const { results, next } = await this.rpc("/discuss/gif/search", params, {
-                silent: true,
-            });
+            const res = await this.sequential(() =>
+                this.rpc("/discuss/gif/search", params, {
```

Comment:
I think (un)setting `loadingGif` should be part of the sequential function, otherwise the first call will reset the value to false (last line of function) and subsequent calls won't reset the value accordingly as it is done outside.

I'm not sure how `position`/`next` is used, but maybe a similar things need to be done, so that we properly take into account the "new" next before calling the rpc?

Reply:
I don't think it's possible? 
sequential returns a Promise that is not resolved, if I do it inside the sequential function, it's not the right timing?

Reply:
The promise is always resolved, the problem is the quick return that was added here below.

Reply:
Right! 
I can just move it before the quick return, it's not an issue

Reply:
I removed the need for a quick return, I think in this case, it's better to avoid any abrupt stop in the middle of the code.